### PR TITLE
feat(pdk): implement private PDK functions to process rate_limiting response headers

### PR DESCRIPTION
**Pull request by ADD-SP**
# REBASE AND MERGE

### Summary

Currently, all rate-limiting-like plugins process their own response headers individually, we should unify this process by a private PDK function.

This PR is also for _[KAG-4679]_.

### Checklist

- [N/A] The Pull Request has tests
- [N/A] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [N/A] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

_[KAG-4679]_

[KAG-4679]: https://konghq.atlassian.net/browse/KAG-4679?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[KAG-4679]: https://konghq.atlassian.net/browse/KAG-4679?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by StarlightIbuki — COMMENTED** [review]
Do we need unit test coverage for this given it's an addition to the PDK?

File: kong/pdk/private/rate_limiting.lua [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+local function _create_rl_ctx(ngx_ctx)
+  assert(not _has_rl_ctx(ngx_ctx), "rate limiting context already exists")
+  local ctx = table_new(0, max_fields_n)
+  ngx_ctx.__rate_limiting_context__ = ctx
+  return ctx
+end
+
+
+local function _get_rl_ctx(ngx_ctx)
```
[the remark this answers is not in the document]
Comment (ADD-SP) [reply kong/pdk/private/rate_limiting.lua:126]: We have discussed this topic offline and not going to do this.

**Comment by ADD-SP** [comment]
> Do we need unit test coverage for this given it's an addition to the PDK?

Added

**Review by StarlightIbuki — APPROVED** [review]
I would do it with busted. But it's fine

**Review by Water-Melon — APPROVED** [review]

**Pull request merged**